app/src/main/java/com/example/pokedex/usecase/GetAllPokemonUseCase.kt

```diff
@@ -0,0 +1,15 @@
+package com.example.pokedex.usecase
+
+import com.example.pokedex.domain.PokemonList
+import com.example.pokedex.repo.PokedexRepository
+import retrofit2.await
+
+interface GetAllPokemonUseCase {
+    suspend fun execute(): PokemonList
```

Comment:
This might be an opportunity to use a flow, which could allow for the use case to manage emiting loading and error states... Pros and cons. Feel free to talk to me about this

Reply:
Done